src/backend/distributed/planner/multi_logical_planner.c

```diff
@@ -515,6 +545,56 @@ HasUnsupportedJoinWalker(Node *node, void *context)
 
 
 /*
+ * ErrorHintRequired returns true if error hint shold be displayed with the
+ * query error message. Error hint is valid only for queries involving reference
+ * and hash partitioned tables. If more than one hash distributed table is
+ * present we display the hint only if the tables are colocated. If the query
+ * only has reference table(s), then it is handled by router planner.
+ */
+static bool
+ErrorHintRequired(char *errorHint, Query *queryTree)
+{
+	List *rangeTableList = NIL;
+	ListCell *rangeTableCell = NULL;
+	List *colocationIdList = NIL;
+
+	if (errorHint == NULL)
+	{
+		return false;
+	}
+
+	ExtractRangeTableRelationWalker((Node *) queryTree, &rangeTableList);
+	foreach(rangeTableCell, rangeTableList)
+	{
+		RangeTblEntry *rte = (RangeTblEntry *) lfirst(rangeTableCell);
+		Oid relationId = rte->relid;
+		char partitionMethod = PartitionMethod(relationId);
+		if (partitionMethod == DISTRIBUTE_BY_NONE)
```

Comment:
This is the first I've seen `DISTRIBUTE_BY_NONE`… I'm assuming it's a reference table thing, but the name is _really_ misleading. @onderkalaci, wasn't there an email thread on what to call this? What is the behavior of it?